### ISSUE TITLE
Issue after updating to 1.7.5.2

### ISSUE THREAD

**Issue by jlorquin** [PrestaShop/PrestaShop#14692]
Hi there,

I updated 1.6.0.14 to 1.7.5.2 and when I click on a product I get the following error in debug mode:

```
(1/1)                                     FatalThrowableError                                                                  Cannot use object of type Currency as array
--
in 759f73aad98c5743169058b96799867a3156c7e9_2.file.product.tpl.php line 101
at Block_19359712305d2e2615901663_33670545->callBlock(object(SmartyDevTemplate))in smarty_internal_runtime_inheritance.php line 248
at Smarty_Internal_Runtime_Inheritance->callBlock(object(Block_19359712305d2e2615901663_33670545), object(SmartyDevTemplate))in smarty_internal_runtime_inheritance.php line 184
at Smarty_Internal_Runtime_Inheritance->process(object(SmartyDevTemplate), object(Block_19359712305d2e2615901663_33670545), object(Block_20530020465d2e261594d459_31529694))in smarty_internal_runtime_inheritance.php line 186
at Smarty_Internal_Runtime_Inheritance->process(object(SmartyDevTemplate), object(Block_20530020465d2e261594d459_31529694))in smarty_internal_runtime_inheritance.php line 156
at Smarty_Internal_Runtime_Inheritance->instanceBlock(object(SmartyDevTemplate), 'Block_20530020465d2e261594d459_31529694', 'head')in d62476ebdc122a9f6ef3329be77dcc99ceb20ce1_2.file.layout-both-columns.tpl.php line 40
at content_5d2e261595a7a7_31179286(object(SmartyDevTemplate))in smarty_template_resource_base.php line 123
at Smarty_Template_Resource_Base->getRenderedTemplateCode(object(SmartyDevTemplate))in smarty_template_compiled.php line 114
at Smarty_Template_Compiled->render(object(SmartyDevTemplate))in smarty_internal_template.php line 216
at Smarty_Internal_Template->render()in smarty_internal_template.php line 385
at Smarty_Internal_Template->_subTemplateRender('layouts/layout-both-columns.tpl', null, 'layouts/layout-full-width.tpl', 0, 3600, array(), 2, false, null, null)in smarty_internal_runtime_inheritance.php line 125
at Smarty_Internal_Runtime_Inheritance->endChild(object(SmartyDevTemplate), 'layouts/layout-both-columns.tpl')in 178e1aab3e0bbdc09b287a31e4fa3777601151c8_2.file.layout-full-width.tpl.php line 42
at content_5d2e2615944f49_35107366(object(SmartyDevTemplate))in smarty_template_resource_base.php line 123
at Smarty_Template_Resource_Base->getRenderedTemplateCode(object(SmartyDevTemplate))in smarty_template_compiled.php line 114
at Smarty_Template_Compiled->render(object(SmartyDevTemplate))in smarty_internal_template.php line 216
at Smarty_Internal_Template->render()in smarty_internal_template.php line 385
at Smarty_Internal_Template->_subTemplateRender('layouts/layout-full-width.tpl', null, 'layouts/layout-full-width.tpl', 0, 3600, array(), 2, false, null, null)in smarty_internal_runtime_inheritance.php line 125
at Smarty_Internal_Runtime_Inheritance->endChild(object(SmartyDevTemplate), 'layouts/layout-full-width.tpl')in 759f73aad98c5743169058b96799867a3156c7e9_2.file.product.tpl.php line 54
at content_5d2e2615934178_99960838(object(SmartyDevTemplate))in smarty_template_resource_base.php line 123
at Smarty_Template_Resource_Base->getRenderedTemplateCode(object(SmartyDevTemplate))in smarty_template_compiled.php line 114
at Smarty_Template_Compiled->render(object(SmartyDevTemplate))in smarty_internal_template.php line 216
at Smarty_Internal_Template->render(false, 0)in smarty_internal_templatebase.php line 232
at Smarty_Internal_TemplateBase->_execute(object(SmartyDevTemplate), null, 'layouts/layout-full-width.tpl', null, 0)in smarty_internal_templatebase.php line 116
at Smarty_Internal_TemplateBase->fetch('catalog/product.tpl', null, 'layouts/layout-full-width.tpl', null, false, true, false)in SmartyDev.php line 40
at SmartyDev->fetch('catalog/product.tpl', null, 'layouts/layout-full-width.tpl')in FrontController.php line 682
at FrontControllerCore->smartyOutputContent('catalog/product.tpl')in FrontController.php line 666
at FrontControllerCore->display()in Controller.php line 301
at ControllerCore->run()in Dispatcher.php line 511
at DispatcherCore->dispatch()in index.php line 28
```

Update log for SQL part: https://pastebin.com/A0yAquMM

PHP version: 7.1.30

I was having errors when updating as explained in [#14683](https://github.com/PrestaShop/PrestaShop/issues/14683) but managed to finish update as as explained [here](https://github.com/PrestaShop/PrestaShop/issues/14617#issuecomment-511739619)

Any help is greatly appreciated. Updating PS is proving a hard task! Thanks!

**Comment by khouloudbelguith** [comment]
Hi @jlorquin,

Thanks for your report.
In my case, I did not manage to reproduce the issue with PS1.7.5.2 after upgrade from PS1.6.1.2.4.
https://drive.google.com/file/d/1IQ6viiNyPLQ7kxg2Nvv8wcRGSTkJ0ynF/view
Could you please try to clear cache manually by deleting all the folder into var/cache (prod & dev).
After enabling debug mode in your store and thanks to debug request which is sent for add new product in your browser, using DevTools, so:

1. enable debug mode
2. go to the product page
3. open up developer tools
4. go to Network tab
5. add product
6. wait for an error, you should see red request information in Network tab
7. click on it, you should see the error message in "Response" or "Preview" tab

Thanks to check and feedback.

**Comment by jlorquin** [comment]
Hi @khouloudbelguith 

I didn't see any red request in the response tab but recorded a [video](https://www.loom.com/share/3200199636bf4100a19d8277234bfa56) in case it helps.

Maybe you misunderstood my issue. I don't have a problem in the back office. It's a front problem.

Thanks for your time.

**Comment by khouloudbelguith** [comment]
@jlorquin, could you please follow this link: https://github.com/PrestaShop/PrestaShop/issues/14595#issuecomment-510884182
& check if your issue is solved.

Thanks!

**Comment by jlorquin** [comment]
@khouloudbelguith, 

I can't access the Localization menu. It takes me to the dashboard.

I've tried other menus and some of them work (taxes, invoices) and some of them either take me to the dashboard or show Acess Denied.

Seems like the update to 1.7 messed everything up!!!!

**Comment by khouloudbelguith** [comment]
@jlorquin, thanks for this clarifications.
I tried to make an upgrade from PS1.6.1.24 to PS1.7.6.0 using the 1-click upgrade v4.7.0. => OK.
There is no redirection to the dashboard page when we click on the International Menu.
https://drive.google.com/file/d/1KMuorMIQPZwATR0gU3g5zqe3h1tpvf-N/view
I tried also with the 1-click upgrade v4.9.0 from PS1.6.1.24 to PS1.7.5.2 (using the local archive)
I manage to reproduce the issue.
I tried to clear cache manually => same issue
https://drive.google.com/file/d/1Aw_QZRXxSb3G5Rq1j9K7PX6L19xGC5FW/view

Ping @Quetzacoalt91 what do you think?
This issue occurs only after an upgrade from PS1.6.1.24 to PS1.7.5.2.

Thanks!

**Comment by jlorquin** [comment]
Thanks @khouloudbelguith,

It seems then that there is an underlying issue that's preventing our system from upgrading successfully.

Are there any further tests we could run?

**Comment by khouloudbelguith** [comment]
@jlorquin, in fact, this issue occurs only after upgrade from PS1.6 to PS1.7.5, but from PS1.6 to PS1.7.6 => OK.
There is no error.

Thanks!

**Comment by LouiseBonnard** [comment]
Hi @khouloudbelguith, you wrote in your last comment that there was no error, should I close this issue?

**Comment by khouloudbelguith** [comment]
Hi @LouiseBonnard, In this [comment](https://github.com/PrestaShop/PrestaShop/issues/14692#issuecomment-512385530), I managed to reproduce the issue **only** `from PS1.6 to PS1.7.5.2.`
I tried again with the 1-click upgrade v4.10.1 from PS1.6 to PS1.7.6.7 => ok.

Thanks!

**Comment by PierreRambaud** [comment]
Does the problem still occurs? Is it related to 1.7.5.2 only? If yes, I think we can close the issue since we don't manage the 1.7.5.2 anymore :thinking:

**Comment by hibatallahAouadni** [comment]
Hello everyone,

I'm closing this issue cause it seems related to 1.7.5.2 only.
As @khouloudbelguith couldn't reproduce it with PS1.7.6.7

@jlorquin Feel free to create a new issue if you encounter any other issue, we will be glad to investigate it.

Thanks!
